### PR TITLE
Update deps and use modern syntax

### PR DESCRIPTION
**Pull request by RomanBurunkov**
Hi @theogravity  this PR contains the following chagnes.

  * Update standard to the latest version.
  * Use const/let instead of var to comply new standard rules.
  * Use arrow funcs.
  * Make code lighter.
  * Use memcache-pp(supports unix sockets, has fewer dependencies) instead of memcache-plus.

The main change is another module for cache client.
It is actually a fork of memcache-plus since it looks a bit outdated and got a lot of old deps.
Also I've made it lighter (get rid of two huge libs) fix some issues, updated some deps and add feature to be able to connect to the cache service via unix sockets. So could you please consider to use it instead of memcache-plus.

Thanks,
Roman.

### CODE REVIEW

**Comment by theogravity** [comment]
The code changes look good, but I have one request. Since you changing the library here, can you refactor it in a way that it takes in the library you want to use? That way this project would not have to depend on a specific library to function as long as the lib you're using shares the same interface, also would mean you wouldn't have to make PRs anymore to do a version bump on the dependent library as well.

For example:

```
function MemcachedClient (options) {
  this.options = options
  if (!this.options) {
    throw new Error('[cache-manager] memcache options not defined')
  }
  
  (!this.options.driver) {
    throw new Error('[cache-manager] memcache driver not specified')
  }

  const Memcached = this.options.driver

  this.memcached = new Memcached(this.options.options)
}
```

**Comment by RomanBurunkov** [comment]
Hi @theogravity ,

It looks that you mean adding a DI to the project.
Sounds good for me, Let put on hold this PR and I will try to implement that.

**Comment by RomanBurunkov** [comment]
Hi @theogravity,

I've added new option `driver` as you suggested earlier.
Add description in readme and also modified tests.

File: package.json
Comment (theogravity) [inline package.json:29]: should be removed

**Review by theogravity — APPROVED** [review]

**Comment by theogravity** [comment]
Thanks - I'll merge and remove the package!

**Comment by theogravity** [comment]
Published as `4.0.0`. 

Thanks for the contribs. I'll add you as a maintainer, and give you npm publish perms.

Please let me know the npm account to add.

**Comment by RomanBurunkov** [comment]
npm `romon2002`

**Comment by theogravity** [comment]
Added, thanks for your help!

**Pull request merged**
